webui/react/src/pages/ExperimentDetails/CodeViewer/IpynbRenderer.tsx

```diff
@@ -1,36 +1,19 @@
+import NotebookJS from '@determined-ai/notebook-render';
 import React from 'react';
-import { CellType, IpynbRenderer } from 'react-ipynb-renderer';
 
-import 'react-ipynb-renderer/dist/styles/onedork.css';
-
-export type IpynbInterface = {
-  cells: CellType[];
-  nbformat: 3 | 4 | 5;
-  worksheets?: {
-    cells: CellType[];
-  }[];
-};
+import './onedork.css';
 
 interface Props {
-  file: IpynbInterface;
+  file: string;
 }
 
 const JupyterRenderer: React.FC<Props> = React.memo(({ file }) => {
   return (
-    <IpynbRenderer
-      bgTransparent={true}
-      formulaOptions={{
-        // katex by default
-        katex: {
-          delimiters: 'gitlab', // dollars by default
-          katexOptions: { fleqn: false },
-        },
-        // optional
-        renderer: 'mathjax',
+    <div
+      className="ipynb-renderer-root"
+      dangerouslySetInnerHTML={{
+        __html: file && NotebookJS.parse(JSON.parse(file)).render().outerHTML,
```

Comment:
do you know whether `NotebookJS` sanitizes? apparently `dangerouslySetInnerHTML` can cause scripts to be executed if not escaped properly.

I tried a quick test by dropping a 
```<p onmouseover='alert(`oops`);'>Hi!</p>```
into the notebook and it did the right thing. 

in the process though i saw that a failure of the JSON parse on this line crashes the page (same thing though happens on [master](http://latest-master.determined.ai:8080/det/experiments/1487/code?filePath=Iris-TensorFlow-Keras.ipynb)

Reply:
@mapmeld Did you test this?

Reply:
According to https://pragmaticwebsecurity.com/articles/spasecurity/react-xss-part2.html the best option for `dangerouslySetInnerHTML` is to use a sanitizer such as dompurify.  I looked and that's what's used by notebookjs: (e.g. in `el.innerHTML = nb.sanitizer(nb.markdown(joined))`)

We're limited in what we can do while still inserting arbitrary notebooks into React.

Reply:
I took a closer look at this; if it's helpful I can remove support for JS and SVG cells which don't do the escape.

Reply:
Let's remove support for JS, you can leave SVG